src/OpenLoco/src/Town.cpp

```diff
@@ -204,4 +210,137 @@ namespace OpenLoco
         }
         return StringIds::town_size_hamlet;
     }
+
+    // 0x00463BD2
+    template<typename Func>
+    static void sub_463BD2(const World::Pos2& topLeftLoc, uint8_t searchSize, Func&& predicate)
```

Comment:
I think we can name this to something like `predicateSearch`? Up to you if you want to leave it for #1742 to tackle.

Reply:
I was thinking maybe boxSearch or squareSearch.

Reply:
Per the ASCII art, I'm thinking `squareSearch` takes the `topLeftLoc` as a corner of the search square. This is reasonably intuitive. Let's go with that then.

Reply:
So turns out my ascii art was wrong. Its still square but i made a mistake in the drawing.